types/types.cc

```diff
@@ -100,15 +100,15 @@ sstring simple_date_to_string(const uint32_t days_count) {
     date::year_month_day ymd{date::local_days{days}};
     std::ostringstream str;
     str << ymd;
-    return str.str();
+    return std::move(str).str();
 }
 
 sstring time_to_string(const int64_t nanoseconds_count) {
     std::chrono::nanoseconds nanoseconds{nanoseconds_count};
     auto time = date::make_time(nanoseconds);
     std::ostringstream str;
     str << time;
-    return str.str();
+    return std::move(str).str();
```

Comment:
instead of using the combination of `ostringstream` and date library, i think a better alternative might be a combination of a plain `std::string` and {fmt}, like
```c++
  std::string s;
  fmt::format_to(std::back_inserter(s), "{:%H:%M:%S}", nanoseconds);
  return s;
```

so we can tighten our dependency a little bit. there are only three places we are using the `date` library.

see https://wandbox.org/permlink/Lo9NUrQNUEqyiMEa and https://godbolt.org/z/YEha9ah7v to compare their outputs.

Reply:
I reverted the change in this particular line. I did not want to change any dependencies via this PR.

Reply:
makes sense. i will post PR to implement this.

Reply:
#17177